Prepare for 2.19 release

matisse opened the pull request:
Change the docker/httpd-2.4/Dockerfile to use apt-get to install
apache and mod_perl instead of building perl from source.

Add README-docker that describes how to use Docker to test this distro.

Fix a COPY directive in docker/httpd-2.2/Dockerfile to work when
invoked from the top level of this distro.

Update README to mention the docker stuff and use metacpan for
finding this distro.

matisse committed:
Prepare for 2.19 release

Change the docker/httpd-2.4/Dockerfile to use apt-get to install
apache and mod_perl instead of building perl from source.

Add README-docker that describes how to use Docker to test this distro.

Fix a COPY directive in docker/httpd-2.2/Dockerfile to work when
invoked from the top level of this distro.

Update README to mention the docker stuff and use metacpan for
finding this distro.

esabol commented:
I would not make these changes, but I really only care about the Perl code being released. So, if they make you happy, it’s fine.

matisse commented:
> I would not make these changes, but I really only care about the Perl code being released. So, if they make you happy, it’s fine.

Hi Ed, can you say why you would not make these changes?

esabol commented:
> Hi Ed, can you say why you would not make these changes?

I think it's useful to be able to test with any version of Perl and mod_perl and not rely on vendor releases. And you got rid of my Makefiles, which I put a lot of effort into and which made building the images and testing easier. But, really, it doesn't matter. I doubt anyone is going to use these Dockerfiles anyway.

matisse commented:
Can you explain what you mean by “  be able to test with any version of Perl and mod_perl and not rely on vendor releases. ”?  And about the Makefiles: I didn’t (intentionally) delete any files. Can you point me to them?

matisse commented:
Ed did you intend to include Makefiles in this commit: https://github.com/matisse/Apache-AuthCookieDBI/commit/fa81b9a7c2a0c59e1bbedeb8812f1e9240823793 ?

esabol commented:
> Can you explain what you mean by “ be able to test with any version of Perl and mod_perl and not rely on vendor releases. ”?

You changed the httpd-2.4/Dockerfile from being able to test any combination of versions of Perl, mod_perl, and Apache httpd to “apt-get install” mod_perl and httpd. That means you are using the vendor (Ubuntu) releases of Perl, mod_perl, and httpd, so you can only test their versions and their releases. Fortunately, the httpd-2.2/Dockerfile still has the capability, so anyone who wants to test other versions can adapt that Dockerfile.

> did you intend to include Makefiles in this commit: fa81b9a ?

Yes, I thought I committed them. I guess I didn’t. I’m not sure how that happened. Anyway, there’s no point now. The README-docker effectively takes their place.

matisse committed:
Update README-docker for other perl/mod_perl/httpd

Update the `README-docker` to point out that the `httpd-2.2/Dockerfile`
can be adapted to test other versions of Perl, mod_perl, and httpd.

matisse commented:
OK thanks for clarifying that (and really for all your work on this.)

FYI: The reason I change the 2.4 Dockerfile (and really I would have change the 2.2 if I could have found vendor packages for it) was speed: the 2.2 file takes 7 times as long to build the image as the apt-get based 2.4 Docker file.


I will update the `README-docker` to point out that the `httpd-2.2/Dockerfile` can be adapted to test other versions of Perl, mod_perl, and httpd.

matisse committed:
Set release date for 2.19
